packages/plugin-hardhat/src/deploy-beacon-proxy.ts

```diff
@@ -72,16 +80,23 @@ export function makeDeployBeaconProxy(hre: HardhatRuntimeEnvironment): DeployBea
     }
 
     const BeaconProxyFactory = await getBeaconProxyFactory(hre, attachTo.signer);
-    const proxyDeployment: Required<ProxyDeployment & DeployTransaction> = Object.assign(
+    const proxyDeployment: Required<ProxyDeployment & DeployTransaction> & DeploymentId = Object.assign(
       { kind: opts.kind },
-      await deploy(BeaconProxyFactory, beaconAddress, data),
+      await deploy(hre, opts, BeaconProxyFactory, beaconAddress, data),
     );
 
     await manifest.addProxy(proxyDeployment);
 
     const inst = attachTo.attach(proxyDeployment.address);
     // @ts-ignore Won't be readonly because inst was created through attach.
     inst.deployTransaction = proxyDeployment.deployTransaction;
+    if (opts.platform && proxyDeployment.deploymentId !== undefined) {
+      inst.deployed = async () => {
+        assert(proxyDeployment.deploymentId !== undefined);
+        await waitForDeployment(hre, opts, inst.address, proxyDeployment.deploymentId);
+        return inst;
+      };
+    }
```

Comment:
Same here regarding updating `deployTransaction`. We might want to define a helper for this entire piece of code.